src/main/java/org/neo4j/sql2cypher/SQLToCypher.java

```diff
@@ -44,38 +44,87 @@
-        System.out.println(new SQLToCypher().convert("""
+        System.out.println(new SQLToCypher().using(new Config()).convert("""
             SELECT t.a, t.b
             FROM my_table AS t
             WHERE t.a = 1
             """));
     }
 
-    // Meta data can be added via .meta(createTable("t").column("a", SQLDataType.INTEGER))
-    private final Parser parser = DSL.using(SQLDialect.DEFAULT).parser();
-    private final Renderer renderer = Renderer.getRenderer(Configuration.prettyPrinting());
-    private final Map<Table<?>, Node> tables = new HashMap<>();
+    private Config config = new Config();
+
+    static class Context {
+        private final Map<Table<?>, Node> tables = new HashMap<>();
+
+        void addTable(Table<?> table, Node node) {
+            this.tables.put(table,node);
+        }
+
+        private Node getNode(Table<?> t) {
+            return tables.get(t);
+        }
+    }
+
+    static class Config {
+        private final SQLDialect dialect;
+        private final Configuration cypherDslConfig;
+
+        public Config() {
+            this.dialect = SQLDialect.DEFAULT;
+            this.cypherDslConfig = Configuration.prettyPrinting();
+        }
+
+        public Config(SQLDialect dialect, Configuration cypherDslConfig) {
+            this.dialect = dialect;
+            this.cypherDslConfig = cypherDslConfig;
+        }
+
+        public SQLDialect getDialect() {
+            return dialect;
+        }
+
+        public Configuration getCypherDslConfig() {
+            return cypherDslConfig;
+        }
+    }
+    public SQLToCypher using(Config config) {
+        this.config = config;
+        return this;
+    }
 
     // Unsure how thread safe this should be (wrt the node lookup table), but this here will do for the purpose of adding some tests
     public String convert(String sql) {
-        Select<?> query = parser.parseSelect(sql);
-        return renderer.render(statement(query));
+        Context ctx = new Context();
+        Select<?> query = parse(sql);
+        ResultStatement statement = statement(query, ctx);
+        return render(statement);
     }
 
-    private ResultStatement statement(Select<?> x) {
+    Select<?> parse(String sql) {
+        Parser parser = DSL.using(config.getDialect()).parser();
+        return parser.parseSelect(sql);
+    }
+
+    String render(ResultStatement statement) {
+        var renderer = Renderer.getRenderer(config.getCypherDslConfig());
+        return renderer.render(statement);
+    }
+
+    ResultStatement statement(Select<?> x, Context ctx) {
 
         // Done lazy as otherwise the property containers won't be resolved
-        Supplier<List<Expression>> resultColumnsSupplier = () -> x.$select().stream().map(t -> (Expression) expression(t)).toList();
+        Supplier<List<Expression>> resultColumnsSupplier =
+                () -> x.$select().stream()
+                        .map(t -> (Expression) expression(t, ctx)).toList();
```

Comment:
Just a question, has this auto formatting been checked in, or how do you folks usually work? I don't have an opinion, other than wanting to adapt to your coding style.

Reply:
Multi line is kinda easier to read.